docs/hopr-documentation/docs/developers/connecting-node.md

```diff
@@ -0,0 +1,248 @@
+---
+id: connecting-node
+title: Interacting with a HOPR node
+---
+
+Once you have your HOPR cluster up and running, you are ready to connect to a single HOPR node and listen to messages sent to it by other HOPR nodes.
+The following guide will show you how to connect to a running HOPR node, and verify some basic functionality using both the REST and WebSocket
+endpoints.
+
+## Requirements before getting started
+
+### 1. Get the node security credentials
+
+To avoid unsecured access to your HOPR node, all HOPR node's WebSocket and REST calls require an `apiToken`. Your API token needs to be appended as a
+`query` parameter for WebSocket connections and as encoded token via the `Authorization` header for you to be able to connect to it. You will not be
+able to connect to your HOPR node without its `apiToken`.
+
+:::info Tip
+The flag used to set this value in a HOPR node via `hoprd` is `--apiToken`. The default `apiToken` used across our documentation is the following one,
+so make sure to change it when running your own node in a public network.
+<br/>
+
+```
+^^LOCAL-testing-123^^
+```
+:::
+
+Make sure to export your `apiToken` to be used in the incoming commands, and every time you open a new terminal.
+
+```
+export apiToken="^^LOCAL-testing-123^^"
+```
+
+### 2. Export your HOPR node REST/WebSocket endpoints
+
+If you followed the ["Running a local HOPR Cluster"](#) section, these will be already exported in your current terminal. Otherwise, you can run
+the following commands to ensure at least your first node's endpoints are exported.
+
+<details>
+  <summary>Export REST and WebSocket endpoints (from local HOPR cluster)</summary>
+  <div>
+    <div>
+    <h3>Node 1</h3>
+    <br/>
+    <pre>
+    export HOPR_NODE_1_HTTP_URL=http://127.0.0.1:13301 HOPR_NODE_1_WS_URL=http://127.0.0.1:19501
+    </pre>
+    <h3>API token & Node 1</h3>
+    <br/>
+    <pre>
+    export apiToken=^^LOCAL-testing-123^^ HOPR_NODE_1_HTTP_URL=http://127.0.0.1:13301 HOPR_NODE_1_WS_URL=http://127.0.0.1:19501
+    </pre>
+    </div>
+    <br/>
+  </div>
+</details>
+
+<details>
+  <summary>Export REST and WebSocket endpoints (from single localhost HOPR node)</summary>
+  <div>
+    <div>
+    <h3>Node 1</h3>
+    <br/>
+    <pre>
+    export HOPR_NODE_1_HTTP_URL=http://127.0.0.1:3001 HOPR_NODE_1_WS_URL=http://127.0.0.1:3000
+    </pre>
+    <h3>API token & Node 1</h3>
+    <br/>
+    <pre>
+    export apiToken=^^LOCAL-testing-123^^ HOPR_NODE_1_HTTP_URL=http://127.0.0.1:3001 HOPR_NODE_1_WS_URL=http://127.0.0.1:3000
+    </pre>
+    </div>
+    <br/>
+  </div>
+</details>
+
+We use `127.0.0.1` instead of `localhost` as some tools like `websocat` (described below) struggle to resolve `localhost` properly.
+
+### 3. Install a WebSocket and REST client
+
+To properly[^1] interact with a HOPR node, you need both a REST and a WebSocket client software. A REST client is used to give instructionst to our
```

Comment:
Minor thing: we do I need BOTH WS and REST client? If I can just send commads via REST and get my success/error responses back I can be happy, right?

Reply:
You do need for receiving messages on the node you are sending messages TO. So yes, if you are interacting with `node 1`, you only need the REST client, but for reading the message `node 1` sends, (e.g. sends a message to `node 2`), you need a WS client connected to `node 2` to see it. I made that comment in the footnote at the end, connected to the word `properly`.